components/brave_wallet_ui/page/screens/swap/utils.ts

```diff
@@ -0,0 +1,193 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+import { BraveWallet, SpotPriceRegistry } from '../../../constants/types'
+import { QuoteOption } from './constants/types'
+
+// Utils
+import Amount from '../../../utils/amount'
+import { getTokenPriceAmountFromRegistry } from '../../../utils/pricing-utils'
+import { makeNetworkAsset } from '../../../options/asset-options'
+
+function getZeroExNetworkFee({
+  quote,
+  fromNetwork
+}: {
+  quote: BraveWallet.ZeroExQuote
+  fromNetwork: BraveWallet.NetworkInfo
+}): Amount {
+  if (!fromNetwork) {
+    return Amount.empty()
+  }
+
+  return new Amount(quote.gasPrice)
+    .times(quote.gas)
+    .divideByDecimals(fromNetwork.decimals)
+}
+
+export function getZeroExFromAmount({
+  quote,
+  fromToken
+}: {
+  quote: BraveWallet.ZeroExQuote
+  fromToken: BraveWallet.BlockchainToken
+}): Amount {
+  return new Amount(quote.sellAmount).divideByDecimals(fromToken.decimals)
+}
+
+export function getZeroExToAmount({
+  quote,
+  toToken
+}: {
+  quote: BraveWallet.ZeroExQuote
+  toToken: BraveWallet.BlockchainToken
+}): Amount {
+  return new Amount(quote.buyAmount).divideByDecimals(toToken.decimals)
+}
+
+export function getZeroExQuoteOptions({
+  quote,
+  fromNetwork,
+  fromToken,
+  toToken,
+  spotPrices,
+  defaultFiatCurrency
+}: {
+  quote: BraveWallet.ZeroExQuote
+  fromNetwork: BraveWallet.NetworkInfo
+  fromToken: BraveWallet.BlockchainToken
+  toToken: BraveWallet.BlockchainToken
+  spotPrices: SpotPriceRegistry
+  defaultFiatCurrency: string
+}): QuoteOption[] {
+  const networkFee = getZeroExNetworkFee({ quote, fromNetwork })
+
+  return [
+    {
+      label: '',
+      fromAmount: new Amount(quote.sellAmount).divideByDecimals(
+        fromToken.decimals
+      ),
+      toAmount: getZeroExToAmount({ quote, toToken }),
+      minimumToAmount: undefined,
+      fromToken,
+      toToken,
+      rate: new Amount(quote.buyAmount)
+        .divideByDecimals(toToken.decimals)
+        .div(new Amount(quote.sellAmount).divideByDecimals(fromToken.decimals)),
+      impact: new Amount(quote.estimatedPriceImpact),
+      sources: quote.sources
+        .map((source) => ({
+          name: source.name,
+          proportion: new Amount(source.proportion)
+        }))
+        .filter((source) => source.proportion.gt(0)),
+      routing: 'split', // 0x supports split routing only
+      networkFee,
+      networkFeeFiat: networkFee.isUndefined()
+        ? ''
+        : networkFee
+            .times(
+              getTokenPriceAmountFromRegistry(
+                spotPrices,
+                makeNetworkAsset(fromNetwork)
+              )
+            )
+            .formatAsFiat(defaultFiatCurrency)
+    }
+  ]
+}
+
+function getJupiterNetworkFee({
+  quote,
+  fromNetwork
+}: {
+  quote: BraveWallet.JupiterQuote
```

Comment:
Do we plan to utilize the quote somehow in the future?

Reply:
Jupiter doesn't include fees in their quote response. Just wanted to be symmetric with the other providers, since the quote usually includes the estimated fee.